### PR TITLE
planner: add the doc for hint

### PR DESCRIPTION
**Pull request by Reminiscent**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`ON`
+- 这个变量用来控制 join method hint 是否会影响 join reorder 的优化过程，包括 `LEADING()` hint 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
+
+> **注意：**
+>
+> v6.7.0 之前的版本行为和该变量设置为 'OFF' 的行为一致。因此当从老版本升级到新版本时一些行为会发生变化，主要是 join 的连接顺序。如果有不符合预期的情况，可以通过将该参数设置为 'OFF' 的方式或者通过使用 `LEADING()` hint 的方式进行规避，强烈推荐使用后一种方式。
```
Comment (songrijie) [inline system-variables.md:2534]: 升级默认保持在`OFF`比较好。

**Comment by songrijie** [comment]
rest LGTM

File: system-variables.md [hunk 2]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`ON`
+- 这个变量用来控制 join method hint 是否会影响 join reorder 的优化过程，包括 `LEADING()` hint 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
+
+> **注意：**
+>
+> v7.0.0 之前的版本行为和该变量设置为 'OFF' 的行为一致，从老版本升级到 v7.0.0 之后版本的集群，为了确保不发生计划变更，该变量会被设置成 'OFF'，但是强烈建议用户尽早将该变量设置成 'ON'。
```
Comment (songrijie) [inline system-variables.md:2534]:
```suggestion
> v7.0.0 之前的版本行为和该变量 'OFF' 的行为一致。为确保向前兼容，从旧版本升级到 v7.0.0 之后版本的集群，该变量会被设置成 'OFF'。为了获取后续版本更灵活的 hint 行为，强烈建议用户在确保无性能回退的情况下，将该变量切换为 'ON'。
```

**Review by songrijie — APPROVED** [review]
LGTM

File: system-variables.md [hunk 3]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`ON`
+- 这个变量用来控制 join method hint 是否会影响 join reorder 的优化过程，包括 `LEADING()` hint 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
+
+> **注意：**
+>
+> v7.0.0 之前的版本行为和该变量 'OFF' 的行为一致。为确保向前兼容，从旧版本升级到 v7.0.0 之后版本的集群，该变量会被设置成 'OFF'。为了获取后续版本更灵活的 hint 行为，强烈建议用户在确保无性能回退的情况下，将该变量切换为 'ON'。
```
Comment (ran-huang) [inline system-variables.md:2534]:
```suggestion
> v7.0.0 之前的版本行为和该变量 `OFF` 的行为一致。为确保向前兼容，从旧版本升级到 v7.0.0 之后版本的集群，该变量会被设置成 `OFF`。为了获取后续版本更灵活的 hint 行为，强烈建议用户在确保无性能回退的情况下，将该变量切换为 `ON`。
```

File: system-variables.md [hunk 4]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
 - 默认值：`OFF`
 - 这个变量用来控制是否强制 inline CTE。默认值为 `OFF`，即默认不强制 inline CTE。注意，此时依旧可以通过 `MERGE()` hint 来开启个别 CTE 的 inline。如果设置为 `ON`，则当前 session 中所有查询的 CTE（递归 CTE 除外）都会 inline。
 
+### `tidb_opt_advanced_join_hint` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`ON`
+- 这个变量用来控制 join method hint 是否会影响 join reorder 的优化过程，包括 `LEADING()` hint 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
```
Comment (ran-huang) [inline system-variables.md:2530]:
```suggestion
- 这个变量用来控制 join method hint 是否会影响 join reorder 的优化过程，包括 [`LEADING()` hint](/optimizer-hints.md#leadingt1_name--tl_name-) 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
```

Comment (ran-huang) [reply system-variables.md:2530]: 这里是否要解释下哪些 hint 属于 join method hint？

Comment (Reminiscent) [reply system-variables.md]: 可以解释，不过有点多，是举个例子还是需要多列出来。

Comment (ran-huang) [reply system-variables.md]: 至少举一个例子吧。optimizer-hint 这篇文档里也没有提过哪些属于 join method hint，可能会有一点定义不清晰

Comment (Reminiscent) [reply system-variables.md]: update. PTAL @ran-huang

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
 - 默认值：`OFF`
 - 这个变量用来控制是否强制 inline CTE。默认值为 `OFF`，即默认不强制 inline CTE。注意，此时依旧可以通过 `MERGE()` hint 来开启个别 CTE 的 inline。如果设置为 `ON`，则当前 session 中所有查询的 CTE（递归 CTE 除外）都会 inline。
 
+### `tidb_opt_advanced_join_hint` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`ON`
+- 这个变量用来控制包括 [`HASH_JOIN` hint](/optimizer-hints.md#hash_joint1_name--tl_name-), [`MERGE_JOIN` hint](/optimizer-hints.md#merge_joint1_name--tl_name-) 等用于控制连接算法的 join method hint 是否会影响 join reorder 的优化过程，包括 [`LEADING()` hint](/optimizer-hints.md#leadingt1_name--tl_name-) 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 join method hint 和 `LEADING()` hint 的场景下会有冲突产生。
```
Comment (Oreoxmt) [inline system-variables.md:2530]:
```suggestion
- 这个变量用来控制包括 [`HASH_JOIN()` Hint](/optimizer-hints.md#hash_joint1_name--tl_name-)、[`MERGE_JOIN()` Hint](/optimizer-hints.md#merge_joint1_name--tl_name-) 等用于控制连接算法的 Join Method Hint 是否会影响 Join Reorder 的优化过程，包括 [`LEADING()` Hint](/optimizer-hints.md#leadingt1_name--tl_name-) 的使用。默认值为 `ON`，即默认不影响。如果设置为 `OFF`，在一些同时使用 Join Method Hint 和 `LEADING()` Hint 的场景下可能会产生冲突。
```

File: system-variables.md [hunk 6]
```diff
@@ -2521,6 +2521,18 @@ mysql> desc select count(distinct a) from test.t;
...
+
+> **注意：**
+>
+> v7.0.0 之前的版本行为和该变量 `OFF` 的行为一致。为确保向前兼容，从旧版本升级到 v7.0.0 之后版本的集群，该变量会被设置成 `OFF`。为了获取后续版本更灵活的 hint 行为，强烈建议用户在确保无性能回退的情况下，将该变量切换为 `ON`。
```
Comment (Oreoxmt) [inline system-variables.md:2534]:
```suggestion
> v7.0.0 之前的版本行为和将该变量设置为 `OFF` 的行为一致。为确保向前兼容，从旧版本升级到 v7.0.0 及之后版本的集群，该变量会被设置成 `OFF`。为了获取更灵活的 Hint 行为，强烈建议在确保无性能回退的情况下，将该变量切换为 `ON`。
```

File: optimizer-hints.md [hunk 7]
```diff
@@ -217,6 +217,32 @@ EXPLAIN SELECT * FROM t WHERE EXISTS (SELECT /*+ SEMI_JOIN_REWRITE() */ 1 FROM t
...
+`SHUFFLE_JOIN(t1_name [, tl_name ...])` 提示优化器对指定表使用 Shuffle Join 算法，该 Hint 只在 MPP 模式下生效。例如：
+
+ ```sql
+ SELECT /*+ SHUFFLE_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id = t2.id;
+ ```
+
+> **注意：**
+>
+> 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
+> 该 Hint 能与 [`HASH_JOIN_BUILD` Hint](#hash_join_buildt1_name--tl_name-) 和 [`HASH_JOIN_PROBE` Hint](#hash_join_probet1_name--tl_name-) 组合使用，达到控制 Shuffle Join 算法的 Build 端和 Probe 端的作用。
```
Comment (Oreoxmt) [inline optimizer-hints.md:231]:
```suggestion
> - 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
> - 该 Hint 能与 [`HASH_JOIN_BUILD` Hint](#hash_join_buildt1_name--tl_name-) 和 [`HASH_JOIN_PROBE` Hint](#hash_join_probet1_name--tl_name-) 组合使用，达到控制 Shuffle Join 算法的 Build 端和 Probe 端的作用。
```

File: optimizer-hints.md [hunk 8]
```diff
@@ -217,6 +217,32 @@ EXPLAIN SELECT * FROM t WHERE EXISTS (SELECT /*+ SEMI_JOIN_REWRITE() */ 1 FROM t
...
+`BROADCAST_JOIN(t1_name [, tl_name ...])` 提示优化器对指定表使用 Broadcast Join 算法，该 Hint 只在 MPP 模式下生效。例如：
+
+ ```sql
+ SELECT /*+ BROADCAST_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id = t2.id;
+ ```
+
+> **注意：**
+>
+> 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
+> 该 Hint 能与 [`HASH_JOIN_BUILD` Hint](#hash_join_buildt1_name--tl_name-) 和 [`HASH_JOIN_PROBE` Hint](#hash_join_probet1_name--tl_name-) 组合使用，达到控制 Broadcast Join 算法的 Build 端和 Probe 端的作用。
```
Comment (Oreoxmt) [inline optimizer-hints.md:244]:
```suggestion
> - 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
> - 该 Hint 能与 [`HASH_JOIN_BUILD` Hint](#hash_join_buildt1_name--tl_name-) 和 [`HASH_JOIN_PROBE` Hint](#hash_join_probet1_name--tl_name-) 组合使用，达到控制 Broadcast Join 算法的 Build 端和 Probe 端的作用。
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Comment by Reminiscent** [comment]
Update. PTAL @Oreoxmt

File: optimizer-hints.md [hunk 9]
```diff
@@ -217,6 +217,32 @@ EXPLAIN SELECT * FROM t WHERE EXISTS (SELECT /*+ SEMI_JOIN_REWRITE() */ 1 FROM t
...
+> - 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
+> - 该 Hint 能与 [`HASH_JOIN_BUILD` Hint](#hash_join_buildt1_name--tl_name-) 和 [`HASH_JOIN_PROBE` Hint](#hash_join_probet1_name--tl_name-) 组合使用，达到控制 Shuffle Join 算法的 Build 端和 Probe 端的作用。
+
+### BROADCAST_JOIN(t1_name [, tl_name ...])
+
+`BROADCAST_JOIN(t1_name [, tl_name ...])` 提示优化器对指定表使用 Broadcast Join 算法，该 Hint 只在 MPP 模式下生效。例如：
+
+ ```sql
+ SELECT /*+ BROADCAST_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id = t2.id;
+ ```
```
Comment (Oreoxmt) [inline optimizer-hints.md:239]:
````suggestion
```sql
SELECT /*+ BROADCAST_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id = t2.id;
```
````

File: optimizer-hints.md [hunk 10]
```diff
@@ -303,6 +329,30 @@ SELECT /*+ HASH_AGG() */ count(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
 SELECT /*+ STREAM_AGG() */ count(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
 ```
 
+### MPP_1PHASE_AGG()
+
+`MPP_1PHASE_AGG()` 提示优化器对指定查询块中所有聚合函数使用一阶段聚合算法，该 Hint 只在 MPP 模式下生效。例如：
+
+ ```sql
+ SELECT /*+ MPP_1PHASE_AGG() */ count(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
+ ```
```
Comment (Oreoxmt) [inline optimizer-hints.md:338]:
````suggestion
```sql
SELECT /*+ MPP_1PHASE_AGG() */ COUNT(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
```
````

File: optimizer-hints.md [hunk 11]
```diff
@@ -303,6 +329,30 @@ SELECT /*+ HASH_AGG() */ count(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
...
+>
+> 使用该 Hint 前，需要保证当前 TiDB 集群能够支持在查询中使用 TiFlash MPP 模式，具体细节见文档[使用 TiFlash MPP 模式](/tiflash/use-tiflash-mpp-mode.md)。
+
+### MPP_2PHASE_AGG()
+
+`MPP_2PHASE_AGG()` 提示优化器对指定查询块中所有聚合函数使用二阶段聚合算法，该 Hint 只在 MPP 模式下生效。例如：
+
+ ```sql
+ SELECT /*+ MPP_2PHASE_AGG() */ count(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
+ ```
```
Comment (Oreoxmt) [inline optimizer-hints.md:350]:
````suggestion
```sql
SELECT /*+ MPP_2PHASE_AGG() */ COUNT(*) FROM t1, t2 WHERE t1.a > 10 GROUP BY t1.id;
```
````

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**